packages/ai/src/ai/account/base.py

```
@@ -48,6 +48,11 @@ class AccountBase(ABC):
     on which edition is active.
     """
 
+    # Server capability tags — subclasses override to declare their mode.
+    # OSS sets ['oss']; SaaS sets ['saas']. Used by the infoOnly auth probe
+    # and copied into every AccountInfo returned by authenticate().
+    capabilities: list[str] = []
```

Ruff (0.15.12)

[warning] Mutable default value for class attribute

(RUF012)